config/externalname.go

```diff
@@ -2233,10 +2233,10 @@ var ExternalNameConfigs = map[string]config.ExternalName{
 	//
 	// Network Firewall Policies can be imported using their ARN
 	// Example: arn:aws:network-firewall:us-west-1:123456789012:firewall-policy/example
-	"aws_networkfirewall_firewall_policy": config.TemplatedStringAsIdentifier("name", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:firewall-policy/{{ .external_name }}"),
+	"aws_networkfirewall_firewall_policy": config.IdentifierFromProvider,
```

Comment:
Is a switch to `config.IdentifierFromProvider` really required here? I understand the deviation mitigation in the `aws_networkfirewall_rule_group` below, but here it [looks](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_firewall_policy#import) static . Does it deviate too or there is another reason that is not obvious? . I any case, please document the decision in the comment above

Reply:
we have the following issue during import - is `.setup.....` not working during import ?

`   Message:               apply failed: creating NetworkFirewall Firewall Policy (firewallpolicy): InvalidRequestException: A resource with the specified name already exists:`

Reply:
I am not sure that the root cause is [this issue](https://github.com/upbound/upjet/issues/32) but sometimes we are observing some interesting issues about the custom external name configurations. This can be also related to these type of problems. And when we switch the custom configurations to `IdentifierFromProvider`, they are fixing.

@ytsarev I think I understand your point. Actually, you say that the ID that we construct is the same as the ID that the provider sends us (IdentifierFromProvider case). I think you are right about this. However, I think that the reason for these types of errors, which cannot be understood well from time to time, may be that the import and apply paths on the terraform side differ for some resources.

In short, we observed similar problems for different resources. We may consider investigate more the root cause of this type of problems.

Maybe @ulucinar also want to say his insights on this.

Reply:
```
he import and apply paths on the terraform side differ for some resources.
```
ah, that might be the case and key for understanding the situation, thank you!

Reply:
Hi folks,
Sorry for the late answer. Just left a comment [here](https://github.com/upbound/provider-aws/issues/657#issuecomment-1507259764) explaining the root causes of the issues we are observing while importing these resources.

I will propose some different fixes than switching to `config.IdentifierFromProvider` because:
- It will introduce breaking changes for both resources
- We will lose `spec.forProvider` API normalization for the `networkfirewall.FirewallPolicy` resource if we switch to `config.IdentifierFromProvider`, i.e., we will introduce the `spec.forProvider.name` field as a naming field for the resource, which's a violation of the XRM.
- We will lose the external-name normalization if we switch to the `config.IdentifierFromProvider` for `aws_networkfirewall_rule_group`: The external-name is right now just the name of the rule group but it will become a full ARN if we make the switch. However, it looks like we need a new feature in upjet's `config.TemplatedStringAsIdentifier`, which I will propose in the review comment suggesting a change. 

Let's discuss, and thank you.